Update dynamics documentation

sveerara opened the pull request:
Moved dynamics documentation from mooseframework wiki page to the new documentation system. 

Fixed bug in DynamicStressDivergenceTensors jacobian.

closes #12087 and #9431.

sveerara committed:
Update dynamics documentation #12087

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/227848/) on 9bff217 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/12088/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/12088/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 74e2d25fd927c4bfe28871af9263a28f4a74b37a`

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/227872/) on 136c88e wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/12088/site)

This comment will be updated on new commits.

bwspenc commented on modules/tensor_mechanics/doc/content/modules/tensor_mechanics/Dynamics.md:
```diff
@@ -0,0 +1,128 @@
+
+## Implementation and Usage
+
+In the moose framework, the mass and stiffness matrices are not explicitly calculated. Only the residuals are calculated. To get the residual, the equation of motion (with Rayleigh damping and HHT time integration) can be written as:
```
moose -> MOOSE

bwspenc commented on modules/tensor_mechanics/doc/content/modules/tensor_mechanics/Dynamics.md:
```diff
@@ -0,0 +1,128 @@
+# Dynamics
+
+Dynamic problems like response of a multi degree of freedom structure to external forcing and wave propagation in a medium can also be solved using the Tensor Mechanics module.
```
'like response' -> 'like the response'

bwspenc commented on modules/tensor_mechanics/doc/content/modules/tensor_mechanics/Dynamics.md:
```diff
@@ -0,0 +1,128 @@
+
+The first two terms to the left ($ \rho \ddot{u}(t+\Delta t) + \eta \rho ((1+\alpha) \dot{u}(t + \Delta t)-\alpha \dot{u}(t)) $) are calculated in the [InertialForce](/InertialForce.md) kernel. $\alpha$, $\beta$, $\gamma$ and $\eta$ need to be passed as input to the InertialForce kernel.
+
+The next two terms to the left involving $\sigma$ are calculated in DynamicStressDivergenceTensors.C.
```
Replace this with a link: `[DynamicStressDivergenceTensors](/DynamicStressDivergenceTensors.md)`

bwspenc reviewed (changes requested):
Looks good.  Just a few minor things to touch up.

sveerara commented:
@bwspenc , thanks for reviewing the PR. I have addressed your comments. I have also added an additional warning stating that DynamicStressDivergenceTensors uses undisplaced mesh whereas InertialForce kernel and Pressure BC use the displaced mesh. 

Bighorn, mammoth and rattlesnake tests are failing. I don't think those tests are failing because of this PR.

moosebuild commented:
Job [Private App tests](https://civet.inl.gov/job/227983/) on 769263c : invalidated by @permcody

Apps updated
